core/vm/instructions.go

```diff
@@ -889,36 +905,40 @@ func makeLog(size int) executionFunc {
 // opPush1 is a specialized version of pushN
 func opPush1(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
 	var (
-		codeLen = uint64(len(scope.Contract.Code))
+		context = scope.RetStack[len(scope.RetStack)-1]
+		codeEnd = context.CodeOffset + context.CodeLength
 		integer = new(uint256.Int)
 	)
-	*pc++
-	if *pc < codeLen {
-		scope.Stack.Push(integer.SetUint64(uint64(scope.Contract.Code[*pc])))
+	*pc += 1
+	dataPos := context.CodeOffset + *pc
+	if dataPos < codeEnd {
+		scope.Stack.Push(integer.SetUint64(uint64(scope.Contract.Code[dataPos])))
 	} else {
 		scope.Stack.Push(integer.Clear())
 	}
 	return nil, nil
 }
 
 // make push instruction function
-func makePush(size uint64, pushByteSize int) executionFunc {
+func makePush(size uint64, pushByteSize uint64) executionFunc {
 	return func(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
-		codeLen := len(scope.Contract.Code)
-
-		startMin := int(*pc + 1)
-		if startMin >= codeLen {
-			startMin = codeLen
+		var (
+			context    = scope.RetStack[len(scope.RetStack)-1]
+			codeEnd    = context.CodeOffset + context.CodeLength
+			pcAbsolute = context.CodeOffset + *pc
+			startMin   = pcAbsolute + 1
+		)
+		if startMin >= codeEnd {
```

Comment:
the conditionals logic here is the exact opposite from go-ethereum one. We have to check if this was on purpose or just never merged from upstream. Maintained the erigon's principle for now.

Reply:
I just move this scope to use upstream logic with ed99df442fffa33e9224e9bc2454578466986d4e because I don't see any reason in the history why it diverged from upstream; the latest meaningful change here was from the EOF work merge commit.

Reply:
Can you have a look at https://github.com/ledgerwatch/erigon/commit/199ede36a2661908343004264ba304fce31a8793#diff-7da47ad076053fe0f266b3088eb9ee206c7f3f8fe5a93116a3f5e25ee5074213.
Both are at Erigon's repo

Reply:
Good find, thanks! 
By my estimation, we can leave the logic as-is (now) without negatively impacting the changes that that PR/patch introduced because 
- my patch (https://github.com/ETCCooperative/erigon/pull/4/commits/ed99df442fffa33e9224e9bc2454578466986d4e)  doesn't change anything with slice allocations
- and the `RightPadBytes` vs. `SetBytes` approach/patch wasn't ultimately adopted; or at least does not seem to be in the original codebase this is on top of
  + https://github.com/ETCCooperative/erigon/pull/4/files#diff-7da47ad076053fe0f266b3088eb9ee206c7f3f8fe5a93116a3f5e25ee5074213L920 (see below for ?)

Something seems wonky; 
- https://github.com/ledgerwatch/erigon/pull/638/files accepts `SetBytes` over `RightPadBytes`, but `RightPadBytes` is still the form at our base ref...?

Reply:
The PR/patch introduction `uint256` seems to be the culprit which reverted the removal of `RightPadBytes` - https://github.com/ledgerwatch/erigon/commit/924c03abf999989373684902540b8b94cbb0a1ad#diff-7da47ad076053fe0f266b3088eb9ee206c7f3f8fe5a93116a3f5e25ee5074213L879-R929.

Reply:
I've just reverted my change that moved the logic to go-ethereum-style; now the function looks as erigon had it originally, but, of course, handling the EOF CodeOffset, CodeLength parameters. 
https://github.com/ETCCooperative/erigon/pull/4/commits/2f4a71742724df4853d67a60a5b767fc17d39733